From d983ead9ab477bbb93dbd1cd0d58ce3fe1adff8a Mon Sep 17 00:00:00 2001
From: Steve Cornelius <steve.cornelius@freescale.com>
Date: Thu, 1 Aug 2013 20:06:05 -0700
Subject: [PATCH 0886/5242] MLK-9710-13 Correct size in BLOB_OVERHEAD
 definition

commit  f50b61373ce1324c6c5f1bbdd83753d3097e2731 from
https://source.codeaurora.org/external/imx/linux-imx.git

Increase the size of BLOB_OVERHEAD.

Signed-off-by: Steve Cornelius <steve.cornelius@freescale.com>
(cherry picked from commit 32aced88dedf1c1c11170d125d3cc546ffefb366)
Signed-off-by: Dan Douglass <dan.douglass@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/crypto/caam/sm.h |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/crypto/caam/sm.h b/drivers/crypto/caam/sm.h
index 2f97f69..51020de 100644
--- a/drivers/crypto/caam/sm.h
+++ b/drivers/crypto/caam/sm.h
@@ -31,7 +31,8 @@
  */
 #define AES_BLOCK_PAD(x) ((x % 16) ? ((x >> 4) + 1) << 4 : x)
 
-#define BLOB_OVERHEAD (16 + 8) /* BKEK + MAC tag storage in any blob */
+/* Define space required for BKEK + MAC tag storage in any blob */
+#define BLOB_OVERHEAD (32 + 16)
 
 /* Keystore maintenance functions */
 void sm_init_keystore(struct device *dev);
-- 
1.7.9.5

